File: solutions/Javascript/Intersection of Two Linked Lists.js
function ListNode(val) {
  this.val = val;
  this.next = null;
}

var getIntersectionNode = function (headA, headB) {
  let addresses = [];
  while (headA || headB) {
    if (headA) {
      if (addresses.indexOf(headA) < 0) {
        addresses.push(headA);
        headA = headA.next;
      } else {
        return headA;
      }
    }
    if (headB) {
      if (addresses.indexOf(headB) < 0) {
        addresses.push(headB);
        headB = headB.next;
      } else {
        return headB;
      }
    }
  }
};

File: solutions/python/Intersection of Two Linked Lists.py
class ListNode:
    def __init__(self, x):
        self.val = x
        self.next = None
        self.isVisited=False

class Solution:
    def getIntersectionNode(self, headA: ListNode, headB: ListNode) -> ListNode:
        while headA:
            headA.isVisited=True
            headA=headA.next
        while headB:
            if hasattr(headB,'isVisited'):
                return headB
            headB=headB.next

